pkg/apis/rollouts/v1alpha1/types.go

```diff
@@ -212,7 +214,9 @@ type RolloutAnalysisStep struct {
 	// TemplateName reference of the AnalysisTemplate name used by the Rollout to create the run
 	TemplateName string `json:"templateName"`
 	// Arguments the arguments that will be added to the AnalysisRuns
-	Arguments []AnalysisRunArgument `json:"arguments,omitempty"`
+	// +patchMergeKey=name
+	// +patchStrategy=merge
+	Arguments []AnalysisRunArgument `json:"arguments,omitempty" patchStrategy:"merge" patchMergeKey:"name"`
 }
 
 // AnalysisRunArgument argument to add to analysisRun
```

Comment:
I'm kinda wondering if we should add a name to canary steps to allow people to override specific steps